mmv1/products/monitoring/Snooze.yaml

```diff
@@ -0,0 +1,96 @@
+  The Snooze will work on alerts that match the criteria defined in the Snooze.
+  The Snooze will be active from interval start time through interval end time.
+
+  ~> **Note:** Monitoring Snoozes cannot be deleted from the Google Cloud Platform.
+  Destroying a Terraform-managed Snooze will cancel the snooze and remove it from state but
+  *will not delete the resource from the project.*
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/monitoring/alerts/snooze'
+  api: 'https://cloud.google.com/monitoring/api/ref_v3/rest#rest-resource:-v3.projects.snoozes'
+base_url: 'v3/projects/{{project}}/snoozes'
+self_link: 'v3/{{name}}'
+create_url: 'v3/projects/{{project}}/snoozes'
+update_verb: :PATCH
+update_mask: true
+id_format: '{{name}}'
+import_format: ['{{name}}']
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  custom_import: templates/terraform/custom_import/self_link_as_name.erb
+  post_create: templates/terraform/post_create/set_computed_name.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    # A snooze can be cancelled but cannot be deleted therefore autogenerated tests fail,
+    # with a dangling resources error. The hand written tests confirm the snooze has been
+    # removed from state.
+    skip_test: true
+    name: 'monitoring_snooze_basic'
+    primary_resource_id: 'snooze'
+    vars:
+      snooze_display_name: 'My Snooze'
+      alert_policy_resource_id: alert_policy
+      alert_policy_display_name: 'My Alert Policy'
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    description: |
+      The unique resource name for this Snooze.
+      Its syntax is: projects/[PROJECT_ID_OR_NUMBER]/snoozes/[SNOOZE_ID]
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      A short name or phrase used to identify the Snooze.
+      The display name is limited to 512 Unicode characters.
+    required: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'criteria'
+    description: |
+      This defines the criteria for applying the Snooze
+    required: true
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: 'policies'
+        item_type: Api::Type::String
+        description: |
+          The specific AlertPolicy names for the alert that should be snoozed.
+          The format is: `projects/[PROJECT_ID_OR_NUMBER]/alertPolicies/[POLICY_ID]`
+          There is a limit of 16 policies per snooze.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'interval'
+    description: |
+      The Snooze will be active from interval.start_time through interval.end_time.
+      The interval.start_time cannot be in the past.
+    required: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: "startTime"
+        description: |
+          The beginning of the time interval. The default value for the start time is the end time.
```

Comment:
It may be simpler to just make this required as well, unless there's a common use case of setting a snooze with the same start and end time.